examples/call_watch/src/main.rs

```diff
@@ -0,0 +1,83 @@
+// Example of watching for new DAI transfers
+// WARNING: This example doesn't account for rollbacks
+
+use hypersync_client::{net_types::Query, CallDecoder, Client, ClientConfig};
+use tokio::time::{sleep, Duration};
+
+const DAI_ADDRESS: &str = "0x6B175474E89094C44Da98b954EedeAC495271d0F";
+
+#[tokio::main]
+async fn main() {
+    env_logger::init().unwrap();
+
+    // create default client, uses eth mainnet
+    let client = Client::new(ClientConfig::default()).unwrap();
+
+    // The query to run
+    let mut query: Query = serde_json::from_value(serde_json::json!( {
+        // start from tip since we only want new transfers
+        "from_block": 20519993,
+        // The logs we want. We will also automatically get transactions and blocks relating to these logs (the query implicitly joins them).
+        "transactions": [
+            {
+                "from": [DAI_ADDRESS],
+            },
+            {
+                "to": [DAI_ADDRESS],
+            },
+        ],
+        // Select the fields we are interested in, notice topics are selected as topic0,1,2,3
+        "field_selection": {
+            "transaction": [
+                "hash",
+                "input",
+            ],
+        }
+    }))
+    .unwrap();
+
+    let decoder = CallDecoder::from_signatures(&["transfer(address dst, uint256 wad)"]).unwrap();
+
+    loop {
+        let res = client.get(&query).await.unwrap();
+
+        for batch in res.data.transactions {
+            for tx in batch {
+                if let Some(decoded_call) = decoder.decode_input(&tx.input.unwrap()).unwrap() {
+                    if decoded_call.len() == 2 {
+                        println!(
+                            "Found DAU transfer {:?}. to: {}, amount: {:?}",
+                            &tx.hash,
+                            decoded_call[0].as_address().unwrap(),
+                            decoded_call[1].as_uint().unwrap()
+                        );
+                    } else if decoded_call.len() == 3 {
+                        println!(
+                            "Found DAU transfer {:?}. from: {}, to: {}, amount: {:?}",
```

Comment:
interesting, how can it decode three values if there are only two arguments in the signature?

Reply:
My bad, I had 2 signatures but accidentally deleted one, will add back